File: benchmarks/computing/montecarlo-pi.js
const monteCarloPi = (numPoints) => {
  let insideCircle = 0

  for (let i = 0; i < numPoints; i++) {
    const x = Math.random()
    const y = Math.random()
    const distance = Math.sqrt(x * x + y * y)

    if (distance <= 1) {
      insideCircle++
    }
  }

  return (insideCircle / numPoints) * 4
}

const numPoints = 100000000 // Adjust the number of points for accuracy vs. speed

console.time("Monte Carlo Pi Calculation")
const calculatedPi = monteCarloPi(numPoints)
console.timeEnd("Monte Carlo Pi Calculation")

console.log(`Approximation of Pi: ${calculatedPi}`)

File: benchmarks/computing/montecarlo-pi.py
import random
import math

def monteCarloPi(numPoints):
    insideCircle = 0

    for i in range(numPoints):
        x = random.random()
        y = random.random()
        distance = math.sqrt(x * x + y * y)

        if distance <= 1:
            insideCircle += 1

    piApproximation = (insideCircle / numPoints) * 4
    return piApproximation

result = monteCarloPi(1000000000)
print("Approximated value of pi:", result)
